hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/AttributeSearchRequestTest.kt

```diff
@@ -0,0 +1,649 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto
+
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jsonMapper
+import com.fasterxml.jackson.module.kotlin.kotlinModule
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.time.LocalDate
+
+class AttributeSearchRequestTest {
+
+  private val objectMapper = jsonMapper { addModules(kotlinModule(), JavaTimeModule()) }
+
+  @Test
+  fun `firstName is John`() {
+    val request = objectMapper.readValue<AttributeSearchRequest>(
+      """{
+            "matchers": [
+              {
+                "joinType": "AND",
+                "textMatchers": [
+                  {
+                    "attribute": "firstName",
+                    "condition": "IS",
+                    "searchTerm": "John"
+                  }
+                ]
+              }
+            ]
+          }
+      """.trimIndent(),
+    )
+
+    assertThat(request).isEqualTo(
+      AttributeSearchRequest(
+        matchers = listOf(
+          Matcher(
+            joinType = JoinType.AND,
+            textMatchers = listOf(
+              TextMatcher(
+                attribute = "firstName",
+                condition = TextCondition.IS,
+                searchTerm = "John",
+              ),
+            ),
+          ),
+        ),
+      ),
+    )
+  }
+
+  @Test
+  fun `firstName is John AND lastName is Smith`() {
+    val request = objectMapper.readValue<AttributeSearchRequest>(
+      """{
+            "matchers": [
+              {
+                "joinType": "AND",
```

Comment:
I didn't want to just follow the format they have currently come up with because it's very much tied to their implementation. This design is flexible and can accomodate changes.